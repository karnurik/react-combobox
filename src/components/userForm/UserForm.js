import React, { Component } from 'react';
import Combobox from '../common/combobox/Combobox'
import './userForm.css';

class UserForm extends Component {
    constructor(props) {
        super(props);
        this.state={
            options : [
                { id: '1', name: 'Candy Eat' },
                { id: '2', name: 'Onions Pick' },
                { id: '3', name: 'Fanny Pick' },
                { id: '4', name: 'Click o Wisp' }],
            userName : '',
            selectedOptions: ['1', '2']
        }
    }

    /* 
      * @description - Function sets the state username for this current controlled element
      * @param event 
      * @return na
    */ 
    handleInputChange = e => {
        this.setState({userName: e.target.value});
    }

    /* 
      * @description - Function is called onchange of the dropdown checkbox.
      * @param optionId 
      * @return na
    */ 
    optionOnchange = (optionId, selectedOptions) => {
        //use these to get currently selected option and current clicked (checked or unchecked) option's id
        console.log('option changed, changed OptionId: ', optionId, selectedOptions)
    }

    render() {
        return (
            <>
                <form  className="form">
                    <label>Username</label>
                    <input type="text" name="userName" placeholder="Username" value={this.state.userName} onChange={this.handleInputChange} required />

                    <Combobox
                        data={this.state.options}
                        placeholder='Select Games'
                        searchPlaceholder='Search Games'
                        label='Games'
                        selectedOptions={this.state.selectedOptions}
                        optionOnchange={this.optionOnchange}
                        />
                
                    <input type="submit" disabled value="Submit" />
                </form>
            </>
        );
    }
}

export default UserForm;