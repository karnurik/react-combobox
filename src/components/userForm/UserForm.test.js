import React from 'react';
import UserForm from './userForm';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';


const options =  [
    { id: '1', name: 'Candy Eat' },
    { id: '2', name: 'Onions Pick' },
    { id: '3', name: 'Fanny Pick' },
    { id: '4', name: 'Click o Wisp' }
];

const userName = 'UjjwalKumar';

test('Component renders successfully', () => {
  const component = renderer.create(
    <UserForm
      options={options}
      userName={userName}
    />
   );
  
  expect(component.toJSON()).toMatchSnapshot();
});

describe('Form suite', () => {

  it('handles input change', () => {
    const wrapper = shallow(<UserForm />);
    wrapper.find('input').first().simulate('change', { target: { value: 'Hello' } });
    expect(wrapper.state().userName).toEqual('Hello')
  })
});