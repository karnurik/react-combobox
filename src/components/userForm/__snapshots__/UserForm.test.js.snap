// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders successfully 1`] = `
<form
  className="form"
>
  <label>
    Username
  </label>
  <input
    name="userName"
    onChange={[Function]}
    placeholder="Username"
    required={true}
    type="text"
    value=""
  />
  <div
    id="outer"
  >
     
    <div
      className="combo-box"
    >
      <label>
        Games
      </label>
      <div
        className="drop-down"
        onClick={[Function]}
      >
        <input
          className="form-input"
          defaultValue="Candy Eat, Onions Pick"
          placeholder="Select Games"
          type="text"
        />
        <i
          className="down-arrow"
        />
      </div>
    </div>
  </div>
  <input
    disabled={true}
    type="submit"
    value="Submit"
  />
</form>
`;
